crates/uv/src/commands/pip/compile.rs

```diff
@@ -103,6 +106,38 @@ pub(crate) async fn pip_compile(
     printer: Printer,
     preview: PreviewMode,
 ) -> Result<ExitStatus> {
+    // If the user requests both `-p` and `--python` or `--python-version`, error
+    if let Some(python_legacy) = python_legacy.as_ref() {
+        if let Some(python) = python.as_ref() {
+            return Err(anyhow!(
+                "Cannot specify both `-p` ({python_legacy}) and `--python` ({python}).",
+            ));
+        }
+        if let Some(python_version) = python_version.as_ref() {
+            return Err(anyhow!(
+                "Cannot specify both `-p` ({python_legacy}) and `--python-version` ({python_version}).",
+            ));
+        }
+    }
+
+    // Respect `UV_PYTHON` with legacy behavior
+    if python_legacy.is_none() && python_version.is_none() && python.is_none() {
+        if let Ok(python) = std::env::var(EnvVars::UV_PYTHON) {
+            if !python.is_empty() {
+                python_legacy = Some(python);
+            }
+        }
+    }
+
+    // Resolve `-p` into `--python-version` or `--python`
+    if let Some(python_legacy) = python_legacy {
+        if let Ok(version) = PythonVersion::from_str(&python_legacy) {
+            python_version = Some(version);
```

Comment:
So the net effect here is that we don't error if the version doesn't exist, right?